Library/ViewModels/CommentCellViewModel.swift

```diff
@@ -85,13 +85,32 @@ public final class CommentCellViewModel:
       .map { comment, _ in comment }
 
     self.authorImageURL = comment
-      .map(\.author.imageUrl)
+      .map {
+        if $0.author.isBlocked {
+          return "" // Use default avatar.
+        } else {
+          return $0.author.imageUrl
+        }
+      }
       .map(URL.init)
-      .skipNil()
 
-    self.body = comment.map(\.body)
+    self.body = comment.map {
+      if $0.author.isBlocked {
+        // TODO(MBL-1026): Use translated string once available.
+        return "This user has been blocked"
+      } else {
+        return $0.body
+      }
+    }
 
-    self.authorName = comment.map(\.author.name)
+    self.authorName = comment.map {
+      if $0.author.isBlocked {
+        // TODO(MBL-1026): Use translated string once available.
+        return "Blocked User"
+      } else {
+        return $0.author.name
```

Comment:
Same comments here, re: putting this directly in the model. Would it make it simpler at all?

Reply:
I don't think this belongs in the model. I think it would complicate things (the model would no longer be encodable which would mess with tests & decoding would be more of a pain). Also, whether a user is blocked or not isn't purely a display thing; it also affects functionality, so it's not like we could do no special handling at all. We also can't consistently always hide blocked user info, since we do want to display usernames of blocked users in messages. Plus, at some point, we might want to add an "unblock" option here, in which case the view model will need to know the real user info.

Reply:
Fair!